### ISSUE TITLE
 Please reenable/reimplement the video rotation button that's missing since the update to 0.20.0

### ISSUE THREAD

**Issue by DrDustwolf** [TeamNewPipe/NewPipe#4500]
### Checklist

- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Please reenable/reimplement the video rotation button. It's missing since the update to 0.20.0

#### Is your feature request related to a problem? Please describe it

It's not possible to easily view a video in fullscreen in Newpipe anymore while laying on one side in bed or on the couch, floor... 

#### How will you/everyone benefit from this feature?

It's a hassle to go into Android settings and deactivate the screen auto rotation just to view a video in full screen and to reenable it again afterwards. The button was a very shrewd feature.

**Comment by avently** [comment]
Maybe I didn't understand something but don't autorotation allows you to hold your phone horizontally and view the video in fullscreen in landscape orientation?

**Comment by nbmrjuhneibkr** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/4478

**Comment by nbmrjuhneibkr** [comment]
I'm pretty sure that this issue is talking about the option to watch videos [like this](https://imgur.com/Q12YjUH), which doesn't exist anymore in v0.20, because manual video rotation control has been removed.

**Comment by B0pol** [comment]
I really don't understand what you mean

> It's not possible to easily view a video in fullscreen in Newpipe anymore while laying on one side in bed

Of course you can

> In 0.20, it looks like this,

Just click on the full screen button:

<img src="https://i.ibb.co/bv1B80f/Screenshot-20201012-124625-New-Pipe.png" width="300">

**Comment by nbmrjuhneibkr** [comment]
You're missing the key point:

>while laying on one side in bed

If you're lying on your side and holding the phone with one hand, you don't want the video to play in landscape mode, because holding a phone in such position would be very inconvenient (let's hope I don't have to take reference photos to get my point across). Screenshot #1 in my comment (taken in v0.18) shows the optimal way to watch a video in such situation, which is no longer possible with v0.20, because someone thought that manual video rotation control should no longer exist in NewPipe.

**Comment by nbmrjuhneibkr** [comment]
And here's another argument in favor of the manual rotation control: there are YouTube videos with 1:1 aspect ratio. Currently NewPipe treats them as landscape videos, and always plays them in landscape mode when you press the full screen button. The video is square, so why does NewPipe have to assume that it should be played in landscape, without giving any choice to the user?

**Comment by Stypox** [comment]
Let's keep discussion in #4383

**Comment by nbmrjuhneibkr** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/4383 is about a bug that unexpectedly puts the whole app in landscape mode.

This is more relevant to https://github.com/TeamNewPipe/NewPipe/issues/4414 , which explicitly mentions "Add the force rotation".

**Comment by opusforlife2** [comment]
Good point, @nbmrjuhneibkr.

@Stypox, since you've made a PR for #4414 that adds the player switch buttons but doesn't add the rotation button, let's keep this one open for that purpose and leave #4414 for the player switch buttons.

**Comment by avently** [comment]
>If you're lying on your side and holding the phone with one hand, you don't want the video to play in landscape mode, because holding a phone in such position would be very inconvenient

You can lock the orientation in portrait via status bar panel. This is what you can do with any apps on a phone. This task shouldn't be made by all apps just because you want it included into the UI. Every feature needs a correct place for it.

This message is not something that needs a reply, it's a final thought that couldn't be changed over time or with a further discussion.

**Comment by nbmrjuhneibkr** [comment]
>It doesn't mean I have a power to say "no" for a feature.

Alright, so the comment expressed a personal opinion which may not represent the opinions of other NewPipe contributors, and shouldn't overrule their decisions either. Thanks for the clarification.  
Comments like that one make this environment appear rather hostile towards anyone who is dissatisfied with major changes introduced by the 0.20 update, and who wants some of them to be optional. So this clarification was necessary.

But what about the misunderstanding of the actual issue, which I pointed out in the first part of my previous comment?

Also, I don't think it's fair to treat requests for features that already existed in NewPipe, but have been removed in the recent update, as other regular feature requests.

**Comment by avently** [comment]
@nbmrjuhneibkr 
>Also, I don't think it's fair to treat requests for features that already existed in NewPipe, but have been removed in the recent update, as other regular feature requests.

It's not about fairness it's about logic.
See, many years ago you could find other features that don't present now. It is because developers removed something and made the app better with other features. If something is removed this is a sign that the feature shouldn't be in the app. In some cases I removed useful features for others and they should be returned but not in the case of this issue.

**Comment by avently** [comment]
@nbmrjuhneibkr 
>Also, I don't think it's fair to treat requests for features that already existed in NewPipe, but have been removed in the recent update, as other regular feature requests.

It's not about fairness it's about logic.
See, many years ago you could find other features that don't present now. It is because developers removed something and made the app better with other features. If something is removed this is a sign that the feature shouldn't be in the app. In some cases I removed useful features for others and they should be returned but not in the case of this issue. 

>But what about the misunderstanding of the actual issue, which I pointed out in the first part of my previous comment?

If you're talking about three points in a list from your top comment then I'm ok with current behaviour. It works like it should. 

Importance level of such points is 1 of 10

**Comment by nbmrjuhneibkr** [comment]
>they should be returned but not in the case of this issue

Again, this doesn't sound like a personal opinion of one contributor, but rather like a final answer for anyone (either user or developer) who thinks that this feature should not have been removed. I'm getting some conflicting messages here.

>If something is removed this is a sign that the feature shouldn't be in the app.

But who gets to decide which features should or shouldn't be removed? A community of developers and users, or a single contributor whose understanding of usage scenarios may be too narrow to make decisions that affect the development in major ways?

After some digging I noticed this comment in your initial PR (which was rejected) for the unified player: https://github.com/TeamNewPipe/NewPipe/pull/834#issuecomment-343673960  
In this comment, original creator of NewPipe points out that manual video rotation control has gained enough recognition to be praised in publications.  
It's clear that this feature was neither unknown nor useless. I've never seen anyone asking for its removal, but I already see comments asking for it to be brought back (which I fully agree with).

**Comment by avently** [comment]
@nbmrjuhneibkr you're asking me and than blame that my opinion is not like yours. Man, i'm not interesting in talking on the issue. You know what I think. Nothing will be changed in my mind even if you give another portion of arguments. If you don't like it it's not my problem.

**Comment by nbmrjuhneibkr** [comment]
No one ever asked *you specifically* to address this issue. Last time I checked, there were other developers who are also working on NewPipe. If you're not interested in it, then why are you even commenting on it?

**Comment by avently** [comment]
@opusforlife2 what about closing this too? Or you think it is a good idea to show the fullscreen button even with autorotation enabled?

**Comment by opusforlife2** [comment]
I don't know WHERE my comments keep vanishing. Somebody help me with my vanishing problem first. 😩 

Anyway, I still think overloading the full screen button to 'force rotate on long press' is a better idea than a separate button. If you're watching a portrait video in full screen, long pressing would turn it to landscape forcefully. And vice versa.

Once you have long pressed on the button to force rotate, all further videos in the queue should obey that orientation.

**Comment by avently** [comment]
@opusforlife2 but this issue is about showing the fullscreen button when autorotation is enabled. So right now a user can rotate the phone to get the orientation rotated. He wants a button which can be clicked to rotate the orientation even with autorotation enabled.
So it's kind of a different issue than what you wrote

**Comment by opusforlife2** [comment]
@avently I have no idea what you're talking about.

This is the title of the issue: "Add back orientation switch button in full screen # 4500"

**Comment by nbmrjuhneibkr** [comment]
>but this issue is about showing the fullscreen button when autorotation is enabled
No, that issue is here: https://github.com/TeamNewPipe/NewPipe/issues/4478

Original description of this issue is kind of confusing, but this part

>It's not possible to easily view a video in fullscreen in Newpipe anymore while laying on one side in bed or on the couch

makes sense if this issue is about

>Adding back orientation switch button in full screen

Which is the title right now, and the author hasn't expressed any complaints about it so far.

**Comment by avently** [comment]
@opusforlife2 sorry, wrote in the wrong issue. I mean #4478. What for you think about closing it?

**Comment by nbmrjuhneibkr** [comment]
>I still think overloading the full screen button to 'force rotate on long press' is a better idea than a separate button. If you're watching a portrait video in full screen, long pressing would turn it to landscape forcefully. And vice versa.

This is nice in theory, but it's also yet another feature that users will not discover without reading about it first. All long press actions require additional explanation.

**Comment by opusforlife2** [comment]
Tooltips are already used for the queuing actions.

**Comment by nbmrjuhneibkr** [comment]
Not everywhere though. I don't remember seeing them in [this](https://user-images.githubusercontent.com/8460563/96157708-4fa9e500-0f1b-11eb-8f40-fdde496916d4.PNG) dialog.

This suggestion is fine for me personally, but I'm not sure if all users will appreciate increasing number of long press actions in the UI, especially for something that was previously available as a dedicated button.

**Comment by opusforlife2** [comment]
Oh. I didn't realise this before, but if you toggle on automatic full screening, you basically wouldn't need the full screen button at all. It could be replaced by force rotation entirely.

**Comment by nbmrjuhneibkr** [comment]
>if you toggle on automatic full screening

This is exactly how it worked before 0.20, but what if you don't want to turn it on? Maybe someone is fine with new player behavior, but just wants to have access to the full screen button without disabling system-wide screen rotation, as described in this issue.

**Comment by nbmrjuhneibkr** [comment]
>the full screen button

I think this is more related to https://github.com/TeamNewPipe/NewPipe/issues/4478

Auto full-screen + manual rotation button is exactly how it worked before 0.20, but what if user is fine with new player behavior (without auto full-screen), and just wants to have access to manual controls?

**Comment by victor-troyan** [comment]
My device is kinda broken, I guess. It is not recognize horizontal orientation anymore, and I can't rotate video manually at all. VLC player and Google Photo. Only official Youtube, which have huge of advertising, can do this. I'm shocked.

**Comment by HattinGokbori87** [comment]
I too am really missing the the "auto full-screen + manual rotation button" feature. Many of us prefer compact and smaller handsets over bigger screen ones. Watching videos in tiny portrait mode is not satisfying experience for (at least) those of us. I don't like the extra hassle to switch videos to full screen for every single short clip. It is an annoying usability issue with YouTube and many of us saw NewPipe's way as a fix. I really hope the devs will bring this feature back.

**Comment by opusforlife2** [comment]
> I too am really missing the the auto full-screen feature.

@HattinGokbori87 Wrong issue. You're looking for #4152.

**Comment by mof22** [comment]
PLEASE restore the full screen reproduction in portrait mode.
In previous versions I can view a landscape video in portrait mode whit the subtitles outside the video frame.
Now subtitles occupy an important part of the video box and frequently obstruct a relevant part of the image. 
This is very useful for those who need the assistance of the subtitles when we do not understand the language of the video. 
In my case I do not understand English when I hear it, but I can read it with some fluidity.  
This is is very important to me because

**Comment by Stypox** [comment]
Note: this has to be discussed along with #4478

**Comment by opusforlife2** [comment]
What the app used to do in 0.19.x: Tap thumbnail -> App checks last stored orientation - > Opens video in that orientation, always in full screen.

What it does now: Tap thumbnail -> Opens video in embedded portrait -> You can go to full screen by tapping icon/rotating phone. Basically, it removes the portrait/landscape distinction in favour of fullscreen (whether portrait or landscape)/not full screen (embedded portrait).

I think the automatic full screen behaviour is incredibly useful in most scenarios, so I would not want to see it go. Whatever solution is adopted should not distort that behaviour. (The full screen video opens in portrait/landscape depending on whatever fills the device screen best.)

What I propose: A "Force Manual Orientation" toggle in Video and Audio settings to override the automatic full screen behaviour. You turn the toggle on -> You get the orientation switch back. Then the app follows the 0.19.x behaviour again. You tap the full screen button -> App checks last stored orientation -> Goes to full screen in that orientation. -> You can tap the button to switch orientation.

**Comment by mof22** [comment]
Let's see if I can explain better. I do not speak English and I'm using what little I know with the help of a translator.

In previous versions at 0.20.0, I could see a landscape video in full screen while the phone was upright. In this way the video appeared at the center of the screen with a black strip up and another under the video.
Why would I want to do this wasting part of the screen? Because in this way the subtitles were out of the picture of the video without obstructing some part of the image.
This is important for me (and perhaps for others) because the subtitles help me understand some language to which my ear is not used to. Some videos even offer automatic translation in subtitles.

**Comment by nbmrjuhneibkr** [comment]
^
This.

And I'd like to add again that it's the most power-efficient way to watch videos on OLED screen, because only ~1/3 of the screen lights up. It also can be good for watching videos is the darkness, when you may not want light up the whole screen right in front of your face for a long time.

**Comment by mof22** [comment]
^
This

I agree.

**Comment by nbmrjuhneibkr** [comment]
> I think the automatic full screen behaviour is incredibly useful in most scenarios, so I would not want to see it go. Whatever solution is adopted should not distort that behaviour.

I agree that newly-implemented features should not be removed _completely_ in favor of restoring 0.19 features. 
Other than the Back button behavior (different issue) - that one is just weird and bad, and would benefit from a full redesign.

> What I propose: A "Force Manual Orientation" toggle in Video and Audio settings to override the automatic full screen behaviour. You turn the toggle on -> You get the orientation switch back. Then the app follows the 0.19.x behaviour again. You tap the full screen button -> App checks last stored orientation -> Goes to full screen in that orientation. -> You can tap the button to switch orientation.

Everything that at least makes it possible to use the app the same way as before is fine by me. But I don't see any real benefits from having a separate override toggle in settings instead of just always displaying one button that switches between portrait/landscape/auto (current behavior) in the player UI.

**Comment by opusforlife2** [comment]
> But I don't see any real benefits from having a separate override toggle in settings instead of just always displaying one button that switches between portrait/landscape/auto (current behavior) in the player UI.

@nbmrjuhneibkr The reason I'm suggesting a toggle is to make it explicitly clear to the user that they are overriding the automatic rotation done by the full screen button.

If you just have the orientation toggle button, then the user could get confused between the automatic behaviour and the manually forced orientations. What would the app remember? The orientation itself? The fact that it needed to be switched in the previous video? It all seems very confusing. The toggle would bypass all these problems, I think.

**Comment by opusforlife2** [comment]
Also, once you use the button to force an orientation, how would you go back to the automatic behaviour? Or will the app not remember orientation, making the user use the button each time they play a video?

If such a situation (like lying sideways in bed) arises rarely, it would be okay to let the user tap the button each time. But @mof22's case is universal. They will need to tap the button each time. So it's better if the orientation is remembered.

**Comment by nbmrjuhneibkr** [comment]
Current automatic orientation switching is bundled with one feature that's actually useful: ability to read comments while the video keeps playing in portrait mode (although there seems to be no way to keep the video on the screen while you're scrolling them, which is a big oversight). I don't think that manual fullscreen orientation control should also eliminate this feature, forcing users to open settings and choose between one or another. Sure, it would be less confusing this way, but also less flexible.



>Also, once you use the button to force an orientation, how would you go back to the automatic behaviour?

Press it again? The button can display one of the 3 different icons depending on the currently selected option.

**Comment by opusforlife2** [comment]
@nbmrjuhneibkr Wouldn't tapping the button again just change the orientation? How do you get out of force mode?

Or are you suggesting the orientation button just switch to whatever the opposite of the current orientation is? And then tapping it again slips the app back into automatic mode, also incidentally changing the orientation? Hmm... That has merit. You could just go back to default orientation that way, yeah.

This would work in the scenario where the user has to select the option for every video. What about saving the orientation across videos? How would that work in this case?

**Comment by nbmrjuhneibkr** [comment]
There are 3 possible modes in total: auto, manual/portrait and manual/landscape. I suggested that instead of having two different options (auto/manual in settings and portrait/landscape in player) it may be easier to have one button that cycles through all 3 modes. Or displays a drop-down menu with 3 options, maybe - I'm not sure what's more convenient.

v0.19 remembers your last choice for portrait/landscape, and applies it to every video until you change it manually. I don't see problems with doing exactly that, but for 3 modes instead of 2.

**Comment by opusforlife2** [comment]
>  it may be easier to have one button that cycles through all 3 modes

Oh... yeah. That makes far more sense than what I was thinking. 😆

I'm for this idea. We already have the case of the Mute button which changes icons upon changing state. It would be similar in case to show a free rotation/rotation lock sort of combo for the 3 states.

**Comment by Stypox** [comment]
I like the idea of a button. But remember that tablets have three possible states: portrait, landscape non-fullscreen and landscape fullscreen. How would that button handle this?

**Comment by SameenAhnaf** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-825470047 isn't the exact same thing. But it will solve the issue and keep the full screen video static, I guess.

**Comment by Fandrelo** [comment]
Before the addition of the unified player, while using the main player on full screen you could press the orientation button and it would force the video in landscape/portrait. This also worked while the video was on "zoom". They should definitely add it back.

**Comment by ChiefMikeK** [comment]
## Just a suggestion
---
![NewPipe_rotation_mockup](https://user-images.githubusercontent.com/20070053/117566400-099bb500-b0e9-11eb-85d0-97b83414b2a8.png)

**Comment by SameenAhnaf** [comment]
@ChiefMikeK That will unnecessarily make a small icon cluttered. 

If video is on 'Landscape' mode: Icon will show as second icon (as an indication for switching to portrait)
If video is on 'Portrait' mode: Icon will show as third icon (as an indication for switching to landscape)
